self-core-impl/src/main/java/com/selfxdsd/core/issues/GithubIssues.java

```diff
@@ -72,4 +104,52 @@ public Issue getById(final int issueId) {
             "You cannot iterate over all the issues in a repo."
         );
     }
+
+    /**
+     * Utility class used for fetching json objects over the network.
+     * @author criske
+     * @version $Id$
+     * @since 0.0.1
+     */
+    private static final class JsonFetcher {
```

Comment:
@criske No utility classes, please. Just implement the call into the json() similarly to how BaseRepo json works.

Reply:
Well I know that, but I thought to use an utility class until we have a network abstraction, we discussed that before. Cause right know we just copy/paste the same code for json objects fetching -  basically introducing the same tight-coupling as an utility class  + DRY violation.
But sure I'll remove that :)

Reply:
Yes, maybe we'll add some networking abstraction later on. But right now, it's better to have a bit of duplicated code (it's code in different objects after all), than having utility classes.

Once someone sees a utility class, the whole concept of encapsulation goes to hell. 

I believe Encapsulation is the most important thing, above all else...